src/node/desktop/src/main/application.ts

```diff
@@ -334,9 +334,23 @@ export class Application implements AppState {
       if (path == null) {
         return exitFailure();
       }
+
+      // a path was selected
       rPath = path;
     }
 
+    // if we don't have an R path at this point, try scanning for R
```

Comment:
I made this change to unravel some of the code flow we had of the form:

    `prepareEnvironment(null)` => `scanForR()` => `prepareEnvironment(<path>)`.

The recursive flow there felt awkward, so I pulled this part out.